arch/Kconfig

```diff
@@ -460,6 +472,12 @@ config ARCH_HAS_EXTRA_EXCEPTION_INFO
 config ARCH_HAS_GDBSTUB
 	bool
 
+config ARCH_USE_CORE_SNIPPETS
+	bool
+
+config ARCH_USE_GNU_LINKONCE
```

Comment:
if `x86_64` is ok to be linked with `.gnu.linkonce.*` this symbol and conditionals in the snippets can be removed